src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -989,14 +1005,7 @@ public bool IsSiloInCluster(SiloAddress silo)
             if (silo.Equals(this.MyAddress)) return true;
 
             var status = this.membershipSnapshot.ClusterMembership.GetSiloStatus(silo);
-            switch (status)
-            {
-                case SiloStatus.Active:
-                case SiloStatus.Joining:
-                    return true;
-                default:
-                    return false;
-            }
+            return status != SiloStatus.None;
```

Comment:
This could have caused incorrect "forwarding to remote cluster" type messages beforehand. It still can (eg before a silo is `Joining` or after it's been cleaned up, but much less likely)